### ISSUE TITLE
Array schema property is not set to required when Not Null is Specified

### ISSUE THREAD

**Issue by Elthis** [ePages-de/restdocs-api-spec#180]
I belive this function causes, that array properties are not added to required in schema (there are two comments that explain what is the issue)
```kotlin
private fun processRemainingSegments(
        builder: ObjectSchema.Builder,
        propertyName: String,
        traversedSegments: MutableList<String>,
        fields: List<JsonFieldPath>,
        propertyField: JsonFieldPath? = null
    ) {
        val remainingSegments = fields[0].remainingSegments(traversedSegments)
        if (remainingSegments.isNotEmpty() && JsonFieldPath.isArraySegment(
                remainingSegments[0]
            )
        ) {
            // there is no adding of required property for array
            traversedSegments.add(remainingSegments[0])
            builder.addPropertySchema(
                propertyName,
                ArraySchema.builder()
                    .allItemSchema(traverse(traversedSegments, fields, ObjectSchema.builder()))
                    .description(propertyField?.fieldDescriptor?.description)
                    .build()
            )
        } else {
            // for object there is
            if (propertyField?.fieldDescriptor?.let { isRequired(it) } == true) {
                builder.addRequiredProperty(propertyName)
            }
            builder.addPropertySchema(
                propertyName,
                traverse(
                    traversedSegments, fields,
                    ObjectSchema.builder()
                        .description(propertyField?.fieldDescriptor?.description) as ObjectSchema.Builder
                )
            )
        }
    }
```

**Comment by ozscheyge** [comment]
Might be related to https://github.com/ePages-de/restdocs-api-spec/issues/90 , too.

**Comment by ooz** [comment]
FIxed with #183
